src/backend/mi2/mi2.ts

```diff
@@ -197,6 +197,10 @@ export class MI2 extends EventEmitter implements IBackend {
 			cmds.push(this.sendCommand("file-exec-and-symbols \"" + escape(target) + "\""));
 		if (this.prettyPrint)
 			cmds.push(this.sendCommand("enable-pretty-printing"));
+		
+		for (const key in this.pathSubstitutions) {
+			cmds.push(this.sendCommand("gdb-set substitute-path " + key + " " + this.pathSubstitutions[key]));
+		};
```

Comment:
this logic would then be moved into GDB, or if this also works with LLDB put this into a function in MI2 which then modifies `this.extraCommands` and is called in gdb.ts before initDebugger

Instead it would here just do
```ts
this.extraCommands.forEach((cmd) => {
	cmds.push(this.sendCommand(cmd));
});
```